### ISSUE TITLE
version 1.3.0 broke VanCone's detection of aria-current page

### ISSUE THREAD

**Issue by b-rad-c** [vanjs-org/van#350]
I was testing the [VanCone](https://github.com/medium-tech/van-cone) example apps against the new version of VanJs and the unit tests are failing. The examples were previously tested against VanJS 1.2.7,  [see example](https://github.com/medium-tech/van-cone/blob/main/examples/spa-app/package.json). 

The feature affects the `link` function [defined here](https://github.com/medium-tech/van-cone/blob/7e5d20b5067dcb90fe3bff5eae332499627face3/src/spa.js#L179). which uses `aria-current` property to change the css class of an `a` tag when `aria-current === 'page'`. And now the unit tests from playwright are failing, saying that the element does not have the correct `aria-current` value.

The problem can be reproduced like this:

- clone the repo: `git clone https://github.com/medium-tech/van-cone`
- `cd examples/spa-app`
- update vanjs in `package.json` `"vanjs-core": "1.2.8"`
- `npm install`
- `npm run test`  >>> unit tests pass
- update vanjs in `package.json` `"vanjs-core": "1.3.0"`
- `rm -R node_modules`
-  `npm install`
- `npm run test`  >>> unit tests fail

**Comment by Tao-VanJS** [comment]
Thanks @b-rad-c for reporting the issue. I am not familiar with playwright test. Could you help me list the steps on how to reproduce the issue in a sample web page? I previewed `examples/spa-app` via `npm run dev` but I couldn't find anything problematic in the web page.

By looking at your source code, this line:

```js
const isCurrentPage = (pageName) => (van.derive(() => currentPage.val === pageName)).val
```

seems suspicious. I am not sure why you wrap the result with a derived state instead of just having the simpler form:

```js
const isCurrentPage = (pageName) => currentPage.val === pageName
```

I suspect the addition wrapper broke the app as in VanJS [`1.3.0`](https://github.com/vanjs-org/van/discussions/275) we made some adjustment to the dependency detection mechanism.

I will dig more if I have a live example that demonstrates where is broken.

**Comment by Tao-VanJS** [comment]
I think I have a complete theory on why things are broken now.

Specifically, the line:

```js
const isCurrentPage = (pageName) => (van.derive(() => currentPage.val === pageName)).val
```

is equivalent to the expanded form:

```js
const isCurrentPage = (pageName) => {
  const a = van.derive(() => currentPage.val === pageName)
  return a.val
}
```

Also, consider the line:

```js
"aria-current": van.derive(() => (isCurrentPage(name) ? "page" : ""))
```

You further declare a new derived state, which calls `isCurrentPage`, which has state `a` as its dependency. Let name the new derived state as `b`.

Now, consider the scenario, `currentPage` changes, which leads to `a` changes, which leads to `b` changes. Note that from the perspective of VanJS, `b` only depends on `a`, as `b` knows nothing about `currentPage`. In VanJS [1.3.0](https://github.com/vanjs-org/van/discussions/275), we slightly modified the dependency detection mechanism to avoid the circular dependency in self-referencing derivations such as:

```js
van.derive(() => ++a.val)
```

The rule is that, if inside a derivation we change a state `a`, then `a` will not be considered as a dependency of the derivation.

My thinking is that declaring a new derived state inside the derivation function is not necessary and should be avoided. I don't see any benefit of it and it will unfortunately confuse the dependency detector.

I think changing `isCurrentPage` to either:

```js
const isCurrentPage = (pageName) => currentPage.val === pageName
```

or

```js
const isCurrentPage = van.derive(currentPage.val === name)
return van.tags.a(
    {
        "aria-current": van.derive(() => (isCurrentPage.val ? "page" : "")),
        ...
    },
    children
);
```

will resolve the issue.

Sorry for the trouble it causes.

**Comment by b-rad-c** [comment]
Thanks for the fast response, the change to `isCurrentPage` has resolved the issue. I will push an updated version of VanCone soon.

Thank you!
